ext/bigdecimal/bigdecimal.c

```diff
@@ -1849,6 +1845,31 @@ BigDecimal_div3(int argc, VALUE *argv, VALUE self)
     return BigDecimal_div2(self, b, n);
 }
 
+ /*
+  *  call-seq:
+  *    add(value, ndigits) -> new_bigdecimal
+  *
+  *  Returns the sum of +self+ and +value+
+  *  with a precision of +ndigits+ decimal digits.
+  *
+  *  When +ndigits+ is less than the number of significant digits
+  *  in the sum, the sum is rounded to that number of digits,
+  *  according to the current rounding mode; see BigDecimal.mode.
+  *
+  *  Examples:
+  *
+  *    BigDecimal('111111.111').add(1, 0) # => 0.111112111e6
```

Comment:
@BurdetteLamar Could you please add the line to specify the rounding mode explicitly before this line?

Reply:
Done.

Reply:
@BurdetteLamar Could you please use `:up` instead of `BigDecimal::ROUND_HALF_UP` because I want to promote using a symbol instead of a constant?

Reply:
Done.  But I have used `:half_up` (the default) because that's how the examples were made.

By the way, I strongly agree that the symbol is better.